draft-ietf-httpbis-origin-frame.md

```diff
@@ -70,80 +75,183 @@ The ORIGIN HTTP/2 frame ({{!RFC7540}}, Section 4) allows a server to indicate wh
 {{!RFC6454}} the server would like the client to consider as members of the Origin Set ({{set}})
 for the connection it occurs within.
 
+## Syntax {#syntax}
+
 The ORIGIN frame type is 0xb (decimal 11).
 
 ~~~~
 +-------------------------------+-------------------------------+
-|         Origin-Len (16)       | Origin? (*)                 ...
+|         Origin-Len (16)       | ASCII-Origin? (*)           ...
 +-------------------------------+-------------------------------+
 ~~~~
 
 The ORIGIN frame's payload contains the following fields, sets of which may be repeated within the
 frame to indicate multiple origins:
 
 Origin-Len:
-: An unsigned, 16-bit integer indicating the length, in octets, of the Origin field.
+: An unsigned, 16-bit integer indicating the length, in octets, of the ASCII-Origin field.
 
 Origin:
 : An optional sequence of characters containing the ASCII serialization of an origin ({{!RFC6454}}, Section 6.2) that the sender believes this connection is or could be authoritative for.
 
-The ORIGIN frame defines the following flags:
+The ORIGIN frame does not define any flags. However, future updates to this specification MAY
+define flags. See {{process}}.
+
+
+## Processing ORIGIN Frames {#process}
+
+The ORIGIN frame is a non-critical extension to HTTP/2. Endpoints that do not support this frame
+can safely ignore it upon receipt.
+
+When received by an implementing client, it is used to initialise and manipulate the Origin Set
+(see {{set}}), thereby changing how the client establishes authority for origin servers (see
+{{authority}}).
+
+The origin frame MUST be sent on stream 0; an ORIGIN frame on any other stream is invalid and MUST
+be ignored.
 
-CLEAR (0x1):
-: Indicates that the Origin Set MUST be reset to an empty set before processing the contents of the frame it occurs upon.
+Likewise, the ORIGIN frame is only valid on connections with the "h2" protocol identifier, or when
+specifically nominated by the protocol's definition; it MUST be ignored when received on a
+connection with the "h2c" protocol identifier.
 
-REMOVE (0x2): 
-: Indicates that the origin(s) carried in the payload must be removed from the Origin Set, if present; if not present, it/they have no effect.
+This specification does not define any flags for the ORIGIN frame, but future updates might use
+them to change its semantics. The first four flags (0x1, 0x2, 0x4 and 0x8) are reserved for
```

Comment:
Well, it has to be somewhere. I did it here because there seem to be some cases that require a breaking change (e.g., if we want to include a DELETE or CLEAR flag down the road), and some that don't (e.g., wildcard support).

Happy to take it out (there are always more ways to do it), but it seemed mostly harmless to try it out here.